tests/scale/threaded_utils/virt.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import annotations
+
+import logging
+import re
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any
+
+from ocp_resources.virtual_machine import VirtualMachine
+from pyhelper_utils.shell import run_ssh_commands
+from timeout_sampler import TimeoutExpiredError, TimeoutSampler
+
+from tests.scale.constants import GUEST_DATA_COMMAND_LIST, GUEST_DATA_RESULT_SEPARATOR
+from utilities.constants import TIMEOUT_1MIN, TIMEOUT_2MIN, TIMEOUT_4MIN, TIMEOUT_8MIN, TIMEOUT_10SEC
+from utilities.virt import (
+    VirtualMachineForTests,
+    wait_for_cloud_init_complete,
+    wait_for_running_vm,
+    wait_for_ssh_connectivity,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def threaded_wait_for_accessible_vms(
+    vms: list[VirtualMachineForTests],
+    timeout: int = TIMEOUT_2MIN,
+    tcp_timeout: int = TIMEOUT_1MIN,
+    sleep: int = TIMEOUT_10SEC,
+) -> list[Any]:
+    assert vms, "No VMs provided {vms!r}"
+
+    def _wait_for_accessible_vm(_vm: VirtualMachineForTests) -> None:
+        wait_for_ssh_connectivity(vm=_vm, timeout=timeout, tcp_timeout=tcp_timeout, sleep=sleep)
```

Comment:
pass `wait_for_ssh_connectivity`
no need for inner func here

Reply:
This inner func is utilized to simplify the the call to `executor.map()` below.
`map()` would require iterables generated for the other arguments